x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/analysis/analyzer/VerifierErrorMessagesTests.java

```diff
@@ -420,46 +420,16 @@ public void testHavingOnScalar() {
                 error("SELECT int FROM test GROUP BY int HAVING 2 < ABS(int)"));
     }
 
-    public void testInWithDifferentDataTypes_SelectClause() {
-        assertEquals("1:17: expected data type [integer], value provided is of type [keyword]",
+    public void testInWithDifferentDataTypes() {
+        assertEquals("1:8: 2nd argument of [1 IN (2, '3', 4)] must be [integer], found value ['3'] type [keyword]",
             error("SELECT 1 IN (2, '3', 4)"));
     }
 
-    public void testInNestedWithDifferentDataTypes_SelectClause() {
-        assertEquals("1:27: expected data type [integer], value provided is of type [keyword]",
-            error("SELECT 1 = 1  OR 1 IN (2, '3', 4)"));
-    }
-
-    public void testInWithDifferentDataTypesFromLeftValue_SelectClause() {
-        assertEquals("1:14: expected data type [integer], value provided is of type [keyword]",
+    public void testInWithDifferentDataTypesFromLeftValue() {
+        assertEquals("1:8: 1st argument of [1 IN ('foo', 'bar')] must be [integer], found value ['foo'] type [keyword]",
             error("SELECT 1 IN ('foo', 'bar')"));
     }
 
-    public void testInNestedWithDifferentDataTypesFromLeftValue_SelectClause() {
```

Comment:
Why removing all these tests?

Reply:
Because this error was caught in the `Verifier` previously, by traversing the plan tree.
Now the check has been moved to the `In` function itself so there's no need to test ORDER BY, WHERE, etc...